geoapps/interpolation/driver.py

```diff
@@ -0,0 +1,295 @@
+#  Copyright (c) 2022 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+
+from __future__ import annotations
+
+import sys
+from os import path
+
+import numpy as np
+from discretize.utils import mesh_utils
+from geoh5py.objects import BlockModel, ObjectBase
+from geoh5py.ui_json import InputFile
+from geoh5py.ui_json.utils import monitored_directory_copy
+from scipy.interpolate import LinearNDInterpolator
+from scipy.spatial import cKDTree
+
+from geoapps.interpolation.params import DataInterpolationParams
+from geoapps.shared_utils.utils import get_locations, weighted_average
+
+
+class DataInterpolationDriver:
+    def __init__(self, params: DataInterpolationParams):
+        self.params: DataInterpolationParams = params
+        self._unique_object = {}
+        self.object_out = None
+
+    def object_base(self, object):
+        for entity in self.params.geoh5.get_entity(object.uid):
+            if isinstance(entity, ObjectBase):
+                return entity
+        return None
+
+    @staticmethod
+    def truncate_locs_depths(locs, depth_core):
+        zmax = locs[:, 2].max()  # top of locs
+        below_core_ind = (zmax - locs[:, 2]) > depth_core
+        core_bottom_elev = zmax - depth_core
+        locs[
+            below_core_ind, 2
+        ] = core_bottom_elev  # sets locations below core to core bottom
+        return locs
+
+    @staticmethod
+    def minimum_depth_core(locs, depth_core, core_z_cell_size):
+        zrange = locs[:, 2].max() - locs[:, 2].min()  # locs z range
+        if depth_core >= zrange:
+            return depth_core - zrange + core_z_cell_size
+        else:
+            return depth_core
+
+    @staticmethod
+    def find_top_padding(obj, core_z_cell_size):
+        pad_sum = 0
+        for h in np.abs(np.diff(obj.z_cell_delimiters)):
+            if h != core_z_cell_size:
+                pad_sum += h
+            else:
+                return pad_sum
+
+    @staticmethod
+    def get_block_model(workspace, name, locs, h, depth_core, pads, expansion_factor):
```

Comment:
This method wouldn't be called at this point. Can we move this method and ansillaries to the utils, until we create the block model creation application?